src/crypto/public_key/elliptic/ecdh.js

```diff
@@ -46,14 +45,13 @@ const webCrypto = util.getWebCrypto();
 const nodeCrypto = util.getNodeCrypto();
 
 // Build Param for ECDH algorithm (RFC 6637)
-function buildEcdhParam(public_algo, oid, cipher_algo, hash_algo, fingerprint) {
-  const kdf_params = new type_kdf_params([hash_algo, cipher_algo]);
+function buildEcdhParam(public_algo, oid, kdfParams, fingerprint) {
   return util.concatUint8Array([
     oid.write(),
     new Uint8Array([public_algo]),
-    kdf_params.write(),
+    kdfParams.replacementKDFParams || kdfParams.write(),
```

Comment:
I'm wondering if this could concatenate the wrong combination for some reason external to this function; for instance `kdfParams.replacementKDFParams` followed by `fingerprint.subarray(0, 20)` instead of `kdfParams.replacementFingerprint`.

Just to add a suggestion to this comment, but do as you see fit:
```
forwarding = kdfParams.replacementKDFParams && kdfParams.replacement

<and then, when concatenating,>

forwarding ? kdfParams.replacementKDFParams : kdfParams.write()
```

Reply:
With @twiss we decided to assume that params are always instantiated properly, otherwise a lot more checks are needed in the kdf params functions. Plus, on encryption the values are set by internal functions, and on decryption the operation fails if someone meddled with them..